### PR TITLE
Add retryMaxCount, overrides retryMaxDuration if provided

### PR DESCRIPTION
**Pull request by bobvanmanen**
Add the `retryMaxCount` option to retry a fixed number of attempts.
The option overrides `retryMaxDuration` when provided

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/retry.js [hunk 1]
```diff
@@ -87,24 +90,33 @@ function filterOptions(options) {
...
  *
  * @param {Integer} interval Min time to wait for retry
- * @param {Boolean} [random=true] Add randomness
  */
-function retryDelay(interval, random = true) {
-    return interval +
-        (random ? Math.floor(Math.random() * 100) : 99);
+function retryDelay(interval) {
+    return interval + Math.floor(Math.random() * 100);
 }
```
Comment (bobvanmanen) [inline lib/retry.js:96]: `random` is never false.

**Review by mfrisbey — APPROVED** [review]

**Comment by bobvanmanen** [comment]
Tested upload and download using the `testbed` application with a bit of instrumentation to ensure the retry options are actually passed in.
Tested download using the `testbed` application, and noted that the retry functionality is working.

File: lib/aem/aemdownload.js [hunk 2]
```diff
@@ -99,12 +99,16 @@ class AEMDownload extends EventEmitter {
             }
         });
 
+        const retryOptions = {
+            retryMaxCount: 5
```
Comment (tmathern) [inline lib/aem/aemdownload.js:103]: Why 5?

**Review by tmathern — APPROVED** [review]

File: lib/aem/aemdownload.js (hunk 2 above)
Comment (bobvanmanen) [reply lib/aem/aemdownload.js:103]: Based on the Nui logs we should at least retry twice, however in this case the code will be running on user desktops on customer networks. So I didn't want to define it too tightly.

**Pull request merged**